src/utils/crypto.ts

```diff
@@ -0,0 +1,148 @@
+import type { IEncryptionResult } from '@/types';
+import { STORAGE_KEYS, PASSWORD_ENCRYPTION_ALGO, IS_EXTENSION } from '@/constants';
+import { WalletStorage } from '@/lib/WalletStorage';
+
+/**
+ * If the application is running as an extension, the key should be extractable.
+ * So that we can save it in the session storage.
+ */
+const IS_EXTRACTABLE = !!IS_EXTENSION;
+const SALT_LENGTH = 16;
+const IV_LENGTH = 16;
+
+const subtleCrypto = globalThis?.crypto?.subtle;
+
+// Utility Functions for Base64 Encoding and Decoding
+const encodeBase64 = (data: Uint8Array) => Buffer.from(data).toString('base64');
+const decodeBase64 = (data: string) => Buffer.from(data, 'base64');
+
+export function generateAndStoreSalt(): Uint8Array {
+  const salt = globalThis.crypto.getRandomValues(new Uint8Array(SALT_LENGTH));
+  WalletStorage.set(STORAGE_KEYS.encryptionSalt, encodeBase64(salt));
```

Comment:
Utility function should not do any side actions like setting local storage state.